test/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/GoodsDestinationControllerSpec.scala

```diff
@@ -18,45 +18,76 @@ package uk.gov.hmrc.merchandiseinbaggagefrontend.controllers
 
   private lazy val view = injector.instanceOf[GoodsDestinationView]
-  private lazy val controller = new GoodsDestinationController(controllerComponents, formProvider, view)
+  private lazy val controller =
+    new GoodsDestinationController(
+      controllerComponents, actionBuilder, formProvider, declarationJourneyRepository, view)
 
   "onPageLoad" must {
-    "return OK and render the view" in {
-      val getRequest = buildGet(routes.GoodsDestinationController.onPageLoad().url)
-      val result = controller.onPageLoad()(getRequest)
+    val url = routes.GoodsDestinationController.onPageLoad().url
+    val getRequest = buildGet(url, sessionId)
+
+    behave like anEndpointRequiringASessionIdAndLinkedDeclarationJourneyToLoad(controller, url)
+
+    "return OK and render the view" when {
+      "a declaration has been started" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney)
+
+        val result = controller.onPageLoad()(getRequest)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual
+          view(form)(getRequest, messagesApi.preferred(getRequest), appConfig).toString
+      }
+    }
+
+    "return OK and render the view" when {
+      "a declaration has been started and a value saved" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney.copy(maybeGoodsDestination = Some(GoodsDestination.values.head)))
+
+        val result = controller.onPageLoad()(getRequest)
 
-      status(result) mustEqual OK
-      contentAsString(result) mustEqual
-        view(form)(getRequest, messagesApi.preferred(getRequest), appConfig).toString
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual
+          view(form.fill(GoodsDestination.values.head))(getRequest, messagesApi.preferred(getRequest), appConfig).toString
+      }
     }
   }
 
   "onSubmit" must {
-    val postRequest = buildPost(routes.GoodsDestinationController.onSubmit().url)
+    val url = routes.GoodsDestinationController.onSubmit().url
+    val postRequest = buildPost(url, sessionId)
+
+    behave like anEndpointRequiringASessionIdAndLinkedDeclarationJourneyToUpdate(controller, url)
 
-    //TODO assert against redirection/storage when MIBM-77 done
     "Redirect to /value-weight-of-goods" when {
-      "a valid selection submitted" in {
+      "a declaration is started and a valid selection submitted" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney)
+
         val request = postRequest.withFormUrlEncodedBody(("value", "ni"))
 
         form.bindFromRequest()(request)
 
         val result = controller.onSubmit()(request)
 
         status(result) mustEqual SEE_OTHER
-        redirectLocation(result).get mustEqual routes.SkeletonJourneyController.valueWeightOfGoods().toString
+        redirectLocation(result).get mustEqual routes.ValueWeightOfGoodsController.onPageLoad().toString
       }
     }
 
     "return BAD_REQUEST and errors" when {
```

Comment:
I wonder if we should/can make this generic as part of  anEndpointRequiringASessionIdAndLinkedDeclarationJourneyToUpdate()

not necessarily as part of this ticket though.